exploratory_analysis/Crypto_Microstructure_BTC_USDT_Exploratory_Analysis.ipynb

```diff
@@ -0,0 +1,4634 @@
+      },
+      "execution_count": null,
+      "outputs": [
+        {
+          "output_type": "execute_result",
+          "data": {
+            "text/plain": [
+              "<Figure size 432x288 with 1 Axes>"
+            ],
+            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXQAAAEICAYAAABPgw/pAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAZ2klEQVR4nO3deZRc5X3m8e/T3erWjgJqYW0gFhkjMkE4fVhie6KYMBbEAxnPBEvxAj6KiU/sjJ044wFMsE2IE08ODnbCTEICZnEMlp2MR7FlY0fAwfEYrAYJg6QRUgRCG1JLSAihpVH3b/6o21KpVXWrulTdrX7r+Zyjo7rv3X73raqnbr23qloRgZmZjXxNw12AmZnVhwPdzCwRDnQzs0Q40M3MEuFANzNLhAPdzCwRDnSzEyTpekn/egLrf1/SdfWsyRqTA92GhKTHJe2W1DaAdULSuYNZ11CT9HlJXy9ui4grI+L+4arJ0uFAt0EnaRbwLiCAq4e1mAoktVTTZnYycqDbUPgw8CRwH3BkaCE7a/+doukjQxeSnsian5W0T9L7s/aPSlov6VVJSyRNK1r/Akk/yuZtl3Rz1t4m6U5JW7N/d/a9U5A0T9JmSf9d0ivA17Kz6G9L+rqkvcD1kk6RdI+kbZK2SLpdUnOpg5X0FUmbJO2V9LSkd2Xt84Gbgfdnx/Rs/36Q1CTpFkkbJe2Q9ICkU7J5s7J3LddJelnSTkmfPeF7x5LhQLeh8GHgH7J/75F0eqUVIuLfZzcvjIjxEfFNSe8G/gy4FpgKbAQeBpA0AfgX4AfANOBcYFm2jc8ClwJzgQuBi4Fbinb3FuBU4EzghqztGuDbwKSs7vuAw9l2LwL+A/A7lLY829epwDeAb0kaHRE/AL4IfDM7pgtLrHt99u/XgLOB8cBf91vmncB5wOXArZLOL1OHNRgHug0qSe+kEJSLI+Jp4N+A365xcx8A7o2IZyLiEHATcFk2pPNe4JWIuCMiDkbE6xHxVNF6t0XEjojoAr4AfKhou73A5yLiUEQcyNp+GhHfiYheYCJwFfCpiHgjInYAfwksKFVkRHw9InZFxOGIuANooxDA1R7jlyNiQ0Tsy45xQb9hny9ExIGIeBZ4lsKLlJkD3QbddcAPI2JnNv0NioZdBmgahbNyALLA2wVMB2ZSeLGouF52e1rRdFdEHOy3zqai22cCo4BtkvZI2gP8LTCl1M4k/ZGkNZJey5Y9BZhc6eByam0Bit/VvFJ0ez+Fs3gzfLHHBo2kMRSGR5qz8WkonK1OknQh8AYwtmiVt1TY5FYK4dq3/XHAacAWCgFc8oy5aL1V2fQZWVufUj85Wty2CTgETI6Iw3kFZuPln6EwHLIqInol7QaUs69StfY5g8JQz3ZgRoV1rcH5DN0G028CPcAcCmPKc4HzgR9TGFdfCbxP0tjs44mL+q2/ncI4cp+HgI9Imptd1Pwi8FREvAR8F5gq6VPZRdAJki4pWu8WSe2SJgO3Asd8dDBPRGwDfgjcIWliduHyHEm/WmLxCRQCuAtokXQrhSGb4mOaJancc+8h4A8knSVpPEfH3HNfSMzAgW6D6zrgaxHxckS80vePwkW+D1AYh+6mEHL3U7j4WOzzwP3ZMMe1EfEvwB8D/whsA84hOyuPiNeBK4D/SGFIYh2FC4sAtwOdwM+B54BnsraB+DDQCqwGdlO4YDq1xHKPULgw+wKF4ZKDHDt8863s/12Snimx/r3Ag8ATwIvZ+r8/wFqtQcl/4MLMLA0+QzczS4QD3cwsEQ50M7NEONDNzBIxbJ9Dnzx5csyaNWu4dm9mNiI9/fTTOyOivdS8YQv0WbNm0dnZOVy7NzMbkSRtLDfPQy5mZolwoJuZJcKBbmaWCAe6mVkiHOhmZomoGOiS7s3+FNbzZeZL0lezPwv2c0lvr3+ZBT29wbI12/nqsnUsW7Odnl7/Do2ZWZ9qPrZ4H4Vfx3ugzPwrgdnZv0uA/5X9X1c9vcGH7nmKlZv2cKC7hzGtzcydOYkHF11Cc5Mqb8DMLHEVz9Aj4gng1ZxFrgEeiIInKfzxglI/K3pCHl+7g5Wb9rC/u4cA9nf3sHLTHh5fu6PeuzIzG5HqMYY+nWN/73lz1nYcSTdI6pTU2dXVNaCdrNq6lwPdPce0HejuYfXWvQMs18wsTUN6UTQi7o6IjojoaG8v+c3Vsi6YNpExrc3HtI1pbWbOtIll1jAzayz1CPQtFP5Ab58ZWVtdzTtvCnNnTkI93RC9jM3G0OedV/Lv9JqZNZx6BPoS4MPZp10uBV7L/gZjXTU3iQcXXUL7un9m0uaf8FcLL/IFUTOzIhU/5SLpIWAeMFnSZuBzwCiAiPgbYClwFbAe2A98ZLCKbW4SY/dsYOyeDVx+/umDtRszsxGpYqBHxMIK8wP4eN0qMjOzmvibomZmiXCgm5klwoFuZpYIB7qZWSIc6GZmiXCgm5klwoFuZpYIB7qZWSIc6GZmiXCgm5klwoFuZpYIB7qZWSIc6GZmiXCgm5klwoFuZpYIB7qZWSIc6GZmiXCgm5klwoFuZpYIB7qZWSIc6GZmiXCgm5klwoFuZpYIB7qZWSIc6GZmiXCgm5klwoFuZpYIB7qZWSIc6GZmiXCgm5kloqpAlzRf0lpJ6yXdWGL+GZIek7RC0s8lXVX/Us3MLE/FQJfUDNwFXAnMARZ
```

Comment:
this is a nice result. actually it says that with 0 fees (big volume), you can get profitable from quoting 7 bps deep on average. and around 15 bps depth, you can get profitable (on smaller volume) even with quite big fees if you would manage your inventory super-well and had very small execution latency.

overall, i agree btc on the biggest exchange is just too competitive for small MM.